src/scheduler/round_robin.rs

```diff
@@ -16,14 +17,25 @@ impl RoundRobinScheduler {
     pub fn new() -> Self {
         Self {
             iterations: 0,
+            max_iterations: 1,
+            data_source: RandomDataSource::initialize(0),
+        }
+    }
+
+    /// Construct a new `RoundRobinScheduler` that will execute the test up to max_iteration times, scheduling its
+    /// tasks in a round-robin fashion.
+    pub fn new_multi_run(max_iterations: usize) -> Self {
```

Comment:
Maybe just collapse this into the existing constructor (which will be API breaking but that's fine)